text/0000-config-hooks.md

```diff
@@ -0,0 +1,73 @@
+- Feature Name: config_hooks
+- Start Date: 10/31/2018
+- RFC PR: (leave this empty)
+- Notion Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Rename `Config` to `Hooks` and allow additional methods for specifying tool resolution.
+
+# Motivation
+[motivation]: #motivation
+
+The settings that are currently called `Configs` within Notion could more accurately be described as Infrastructure Hooks. They define settings that, as opposed to being individual user configurations, will most likely be defined across an entire organization:
+
+- Alternate locations for tool indexes
+- Alternate URLs for resolving specific tool versions
+- Event handling for tracking performance metrics
+
+To prevent confusion and naming conflicts with eventual user configurations, these settings should be renamed. Also, currently the only implemented settings are resolving tool versions using a shell command. It would be more useful if there were additional methods of customizing both the tool resolution and tool index, including but not limited to:
+
+- Static URL Prefix
+- URL Template with Predefined Variables
+- (Existing) Shell Command
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+We will need to define the allowed resolution methods and the structure of the relevant `.toml` file. We will also need to define the template format and the variables that are available for use in the templates. We should also have documentation of the API for the Shell Command, both the arguments that will be passed to it as well as the expected output.
+
+# Details
+[details]: #details
+
+## Rename
+
+This change is relatively straightforward, renaming from `Config` to `Hooks` internally, as well as in any documentation. It should also include switching the file from which these settings are loaded from `config.toml` to `hooks.toml`.
+
+## URL Prefix Resolver
+
+This option should store the user-defined URL prefix and then use the default behavior for determining the variable part of the URL in any resolution.
```

Comment:
Will we need to support multiple URL prefixes for this case, depending on how the repository is organized?

Reply:
My thinking was that the URL Prefix would be the more simplistic option for cases where it works. Each of the hooks would be configurable per-tool and per-use (one use being the tool version index and the other being the binary file resolution path), so we would support different URL Prefixes for different tools and uses, but for a given tool use case only a single prefix would be supported.

If they need more involved paths then that can be supported with the URL Template hook.